### ISSUE TITLE
rancher2_app_v2 values not passed to sub charts (part 2)

### ISSUE THREAD

**Issue by Sartigan** [rancher/terraform-provider-rancher2#813]
I don't know if it is similar to: https://github.com/rancher/terraform-provider-rancher2/issues/545 Everything but the "global" section is passed to the app_v2 values.

**Provider version:** rancher/rancher2 v1.21.0
**Terraform Version:** 1.0.0

```
  + resource "rancher2_app_v2" "rancher_monitoring" {
      + annotations                 = (known after apply)
      + chart_name                  = "rancher-monitoring"
      + chart_version               = "100.0.0+up16.6.0"
      + cleanup_on_fail             = false
      + cluster_id                  = "local"
      + cluster_name                = (known after apply)
      + disable_hooks               = false
      + disable_open_api_validation = false
      + force_upgrade               = false
      + id                          = (known after apply)
      + labels                      = (known after apply)
      + name                        = "rancher-monitoring"
      + namespace                   = "cattle-monitoring-system"
      + project_id                  = "local:p-wrz89"
      + repo_name                   = "rancher-charts"
      + system_default_registry     = (known after apply)
      + values                      = <<-EOT
            defaultRules:
              create: false
              labels:
                severity: warning
            global:
              cattle:
                clusterId: local
                clusterName: local
                systemDefaultRegistry: nexus3.com
        EOT
```

When I check the values.yaml from the app:

```
global:
  cattle:
    systemDefaultRegistry: ''
    windows:
      enabled: false
  imagePullSecrets: []
  kubectl:
    pullPolicy: IfNotPresent
    repository: rancher/kubectl
    tag: v1.20.2
  rbac:
    create: true
    pspAnnotations: {}
    pspEnabled: true
    userRoles:
      aggregateToDefaultRoles: true
      create: true
```

It is like the `global` section is ignored

**Comment by rawmind0** [comment]
The global values should be passed in the same way to the app_v2, not able to reproduce the issue.

The `global.cattle.clusterId|clusterName|systemDefaultRegistry` are injected by the provider, you don't need to add them to the app_v2 values. The `clusterId` is got from `rancher2_app_v2. cluster_id ` and then got the `clusterName`. The `systemDefaultRegistry` is got from the Rancher settings, so to use `nexus3.com` you'd need to configure it on Rancher `Global Settings` before.

Also, i think you are not checking the applied values properly. I guess you are checking the `spec.chart.values` field, but the proper one would be `spec.values`, where you can see the `global.cattle` data. Could you please confirm that?

**Comment by rawmind0** [comment]
Please, reopen the issue if still needed.

**Comment by spielkind** [comment]
Hey @rawmind0, we are facing the same issue.

1. We changed the systemDefaultRegistry at Rancher global settings.
2. We changed the values.yaml (global.cattle.systemDefaultRegistry), which is useless, when using terraform as I learned.
3. Rerun our TF pipeline, but the helm deployment didn't got changed.

We tried this with rancher-monitoring, but seems to be the same for at least rancher-logging.

So I guess the "global settings" systemDefaultRegistry is only working when installing a new app, not when changing/upgrading an already installed app to a new systemDefaultRegistry?

At https://<rancher_url>/v3/settings/system-default-registry we get the new registry.
